tgui/packages/tgui/interfaces/common/RecipeLookup.js

```diff
@@ -0,0 +1,210 @@
+          </LabeledList.Item>
+        )}
+        <LabeledList.Item bold label="Purity">
+          <LabeledList>
+            <LabeledList.Item label="Optimal pH range" >
+              <Button
+                color="transparent"
+                textColor="white"
+                tooltipPosition="right"
+                content={recipe.lowerpH + "-" + recipe.upperpH}
+                tooltip="If your reaction is kept within these bounds then the purity of your product will be 100%" />
+            </LabeledList.Item>
+            <LabeledList.Item label="Inverse purity">
+              <Button
+                color="transparent"
+                textColor="white"
+                tooltipPosition="right"
+                content={recipe.inversePurity}
+                tooltip="If your purity is below this it will 100% convert into the product's associated Inverse reagent on consumption." />
+            </LabeledList.Item>
+            <LabeledList.Item label="Minimum purity">
+              <Button
+                color="transparent"
+                tooltipPosition="right"
+                textColor="white"
+                content={recipe.minPurity}
+                tooltip="If your purity is below this at any point during the reaction, it will cause negative effects, and if it remains below this value on completion it will convert into the product's associated Failed reagent." />
+            </LabeledList.Item>
+          </LabeledList>
+        </LabeledList.Item>
+        <LabeledList.Item bold label="Rate profile" width="10px">
+          <Flex
+            height="50px"
+            width="225px"
+            position="relative"
+            style={{
+              'background-color': 'black',
+            }} >
+            <Chart.Line
+              fillPositionedParent
+              data={recipe.thermodynamics}
+              strokeWidth={0}
+              fillColor={"#3cf072"} />
+            {recipe.explosive && (
+              <Chart.Line
+                position="absolute"
+                top={0.01}
+                bottom={0}
+                right={recipe.isColdRecipe?16.41:0}
+                width="28px"
+                data={recipe.explosive}
+                strokeWidth={0}
+                fillColor={"#d92727"} />
+            )}
+          </Flex>
+        </LabeledList.Item>
+        <LabeledList.Item bold label="Dynamics">
+          <Flex position="relative" top="5px" left="-12px">
+            <Button
+              color="transparent"
+              textColor={recipe.isColdRecipe ? "red" : "white"}
+              content={recipe.isColdRecipe ? recipe.explodeTemp+"K" : recipe.tempMin+"K"}
+              tooltip={recipe.isColdRecipe ? "The temperature at which it is underheated, causing negative effects on the reaction." : "The minimum temperature needed for this reaction to start. Heating it up past this point will increase the reaction rate."}
```

Comment:
Break up into multiple lines. Linter ignores these lines because they have strings, but it doesn't exempt you from at least putting newlines before `?` and `:`.

Reply:
Didn't realise I could do that! Thanks.